lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +36,51 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._validate_dataset()
+        self._storage = {}
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError('File does not exists')
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError('Path does not lead to directory')
+
+        meta_files = list(self.path_to_raw_txt_data.glob("*_meta.json"))
+        raw_files = list(self.path_to_raw_txt_data.glob("*_raw.txt"))
+
+        if len(meta_files) != len(raw_files):
+            raise InconsistentDatasetError("Number of meta and raw files is not equal")
+
+        if not meta_files or not raw_files:
+            raise EmptyDirectoryError("Directory is empty")
+
+        file_ids = [int(file.name.split("_")[0]) for file in raw_files]
+        if len(file_ids) != len(set(file_ids)):
+            raise InconsistentDatasetError("IDs contain duplicates")
```

Comment:
this error should also be raised if numeration is incorrect

Reply:
added